### PR TITLE
Doc updates for dnsTunneling and autoProxy

### PR DESCRIPTION
**Pull request by CatalinFetoiu**
Updates:

in wsl-config.md - change the defaults of dnsTunneling and autoProxy to true
                            - add the "dnsTunnelingIpAddress" experimental setting

in troubleshooting.md - add troubleshooting notes for autoProxy and dnsTunneling

### CODE REVIEW

**Review by craigloewen-msft — APPROVED** [review]

**Review by keith-horton — APPROVED** [review]

File: WSL/troubleshooting.md [hunk 1]
```diff
@@ -275,16 +277,16 @@ When WSL can’t connect to the internet, it might be because the DNS call to th
...
 - WSL will use Windows DNS suffixes in the following order (similar to the order used by the Windows DNS client): 
   1. Global DNS suffixes 
   2. Supplemental DNS suffixes 
   3. Per-interface DNS suffixes
   4. If DNS encryption (DoH, DoT) is enabled on Windows, encryption will be applied to DNS queries from WSL. If users want to enable DoH, DoT inside Linux, they need to disable DNS tunneling.
-- DNS queries from Docker containers (either Docker Desktop or native Docker running in WSL) will bypass DNS tunneling. DNS tunneling cannot be leveraged to apply host DNS settings and policies to Docker DNS traffic.
-- Docker Desktop has its own way (different from DNS tunneling) of applying host DNS settings and policies to DNS queries from Docker containers.
+- DNS queries from Docker containers managed by Docker Desktop will bypass DNS tunneling. DNS tunneling cannot be leveraged to apply host DNS settings and policies to those Docker containers. Docker Desktop has its own way (different from DNS tunneling) of applying host DNS settings and policies to DNS queries from Docker containers.
```
[the remark this answers is not in the document]
Comment (CatalinFetoiu) [reply WSL/troubleshooting.md]: resolved in commit 5. thanks for the observation

File: WSL/troubleshooting.md [hunk 2]
```diff
@@ -267,6 +267,8 @@ When enabled, the following apply to proxy settings on your Linux distributions:
 - The Linux environment variable, `NO_PROXY`, is set to bypass any HTTP/S proxies found in the Windows configuration targets.
 - Every environment variable, except `WSL_PAC_URL`, is set to both lower case and upper case. For example: `HTTP_PROXY` and `http_proxy`.
 
+There is a known issue caused by ZScaler configurations, where ZScaler repeatedly enables and disables Windows proxy configurations, leading to WSL repeatedly showing the "An Http proxy change has been detected on the host" notification.
```
Comment (mattwojo) [inline WSL/troubleshooting.md:270]: Can we expect that our target audience here will know what ZScaler configurations are? Should we explain any further or provide a link?

**Review by mattwojo — APPROVED** [review]
Looks good! Thanks for the update. One question about whether to explain ZScaler Configurations at all.

File: WSL/troubleshooting.md (hunk 2 above)
Comment (CatalinFetoiu) [reply WSL/troubleshooting.md:270]: the issue that motivated this note was reported at https://github.com/microsoft/WSL/issues/11523

we know the issue is caused by ZScaler but we don't know how ZScaler was configured

should we rephrase this to "caused by ZScaler" ? and we can add the link to the WSL github issue above
